MyHashMap.java

```diff
@@ -0,0 +1,228 @@
+                                return currNode.value;
+                            }
+                        }
+                    }
+                }
+                sizeArr *= 2;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void clear() {
+        loadFactor = 0.75;
+        threshold = 12;
+        size = 16;
+        capacity = 0;
+        nodesArr = new MyHashMap.Node[size];
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public V get(K key) {
+        if (key == null) {
+            if (nodesArr[0].key == null) {
+                return nodesArr[0].value;
+            } else {
+                Node currNode = nodesArr[0].next;
+                if (currNode != null) {
+                    if (currNode.key == null) {
+                        return currNode.value;
+                    }
+                    while (currNode.key != null) {
+                        currNode = currNode.next;
+                        if (nodesArr[0].key == null) {
+                            return currNode.value;
+                        }
+                    }
+                }
+            }
+        } else {
+            int sizeArr = 16;
+            while (sizeArr <= size) {
+                int index = hashIndex(key, sizeArr);
+                if (nodesArr[index] != null) {
+                    if (nodesArr[index].key.equals(key)) {
+                        return nodesArr[index].value;
+                    } else {
+                        Node currNode = nodesArr[index].next;
+                        if (currNode.key.equals(key)) {
+                            return currNode.value;
+                        }
+                        while (currNode.next != null) {
+                            currNode = currNode.next;
+                            if (nodesArr[index].key.equals(key)) {
+                                return currNode.value;
+                            }
+                        }
+                    }
+                }
+                sizeArr *= 2;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder stringValue = new StringBuilder("");
+        for (int i = 0; i < nodesArr.length; i++) {
+            try {
+                Node currentNode = nodesArr[i];
+                stringValue.append(currentNode.key).append(" ").append(nodesArr[i].value).append("\n");
+                while (currentNode.next != null) {
+                    currentNode = currentNode.next;
+                    stringValue.append(currentNode.key).append(" ").append(currentNode.value).append("\n");
+                }
+            } catch (Exception e) {
+            }
+        }
+        return stringValue.toString();
+    }
+
+    class Node {
+        int hashcode;
+        K key;
+        V value;
+        Node next;
+
+        public Node getNode(int a) {

```

Comment:
Constructor should not return null, what is the purpose of it?

Reply:
> Constructor should not return null, what is the purpose of it?

oops, glee code, which I forgot to remove, my guilt